### PR TITLE
Added algorithm for Doubly Linked List in C

### LINKED ISSUE #358: [Feature Request] Doubly Linked List
## 🚀 Feature Request
To add the algos of Doubly Linked list and circular linked list

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No  
@ajay-dhangar Kindly assign me .

### PR DESCRIPTION
**Pull request by Mansi07sharma**
### Description
Added a Doubly linked list in C Language

Fixes # 358

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/linked-list/Doubly-linked-list-C.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
+---
+id: DoublyLinkedList
```
Comment (ajay-dhangar) [inline docs/linked-list/Doubly-linked-list-C.md]: replace `id: DoublyLinkedList` with `id: doubly-linked-list`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Mansi07sharma** [comment]
@ajay-dhangar done changes.

**Review by shivay-coding — APPROVED** [review]
Thank you for your contributions to Algo. Your work is good, and I appreciate your efforts.

**Pull request merged**